internal/engine/upper_test.go

```diff
@@ -1002,6 +1002,11 @@ func (f *testFixture) testPodWithDeployID(podID string, manifestName string, pha
 	}
 }
 
+func setImage(pod *v1.Pod, image string) {
+	pod.Spec.Containers[0].Image = image
```

Comment:
i wonder if we should have an assertion that this is a special tilt tag (like "tilt-abcdef")

Reply:
...huh? can you say more? how would this be helpful?

Reply:
we want our tests to accurately mock out production behavior

Reply:
we might want to test pod events that have an image without a tilt tag, so i'm gonna not add that for now

Reply:
👍